new/analyse/spec/add-dependencies.spec.ts

```diff
@@ -0,0 +1,103 @@
+import type { Env, Module, Package } from '../../types';
+import { envWithConfig } from '../../fixtures/env';
+
+// mock Packages
+const mockPackageBuilder: (path: string, name: string) => Package = (
+  path,
+  name
+) => ({
+  path,
+  packageJson: { name },
+  messages: [],
+  relatedMessages: [],
+  determinedIncrementLevel: -1,
+  dependers: {
+    dependencies: [],
+    devDependencies: [],
+    peerDependencies: [],
+  },
+  dependees: {
+    dependencies: [],
+    devDependencies: [],
+    peerDependencies: [],
+  },
+});
+
+const mockEnvPackages: Env['packages'] = {
+  mock1Package: mockPackageBuilder('path/to/1/', 'mock1Package'),
+  mock2Package: mockPackageBuilder('path/to/2/', 'mock2Package'),
+  mock3Package: mockPackageBuilder('path/to/3/', 'mock3Package'),
+  mock4Package: mockPackageBuilder('path/to/4/', 'mock4Package'),
+};
+mockEnvPackages.mock1Package.packageJson.dependencies = {
+  lodash: '1',
+  mock2Package: '1',
+};
+mockEnvPackages.mock1Package.packageJson.devDependencies = {
+  eslint: '1',
+  mock3Package: '1',
+};
+mockEnvPackages.mock1Package.packageJson.peerDependencies = {
+  react: '1',
+  mock4Package: '1',
+};
+
+// mock Logger
+const mockError = jest.fn();
+const mockLog = jest.fn();
+const mockLogger = jest.fn(() => ({ error: mockError, log: mockLog }));
+jest.doMock('../../helpers/logger', () => ({ logger: mockLogger }));
+
+describe('addDependencies Module', () => {
+  let addDependencies: Module;
+  beforeAll(() => {
+    addDependencies = require('../add-dependencies').addDependencies;
+  });
+
+  it('Sets up a logger', () => {
+    expect(mockLogger).toBeCalledTimes(1);
+    expect(mockLogger).toBeCalledWith('add dependencies');
```

Comment:
They're the easier piece of each .spec file I've created 😅  they're super lightweight to test.
I think it won't hurt to have them covered, or at least mock all logger modules out

I'd insist on keeping the error ones, if we decide to go for error logs instead of throwing exceptions in the future. In case of problems, they'll help the user to know what went wrong and how to report problems to us in the future

Reply:
Yepp ... The error ones are important, I guess. For the others I'd opt for mocking them out. In the end we anyway might want to optimise the output and probably create a more verbose variant for debugging.